tests/Aspire.Hosting.Redis.Tests/RedisFunctionalTests.cs

```diff
@@ -50,6 +52,77 @@ public async Task VerifyRedisResource()
         Assert.Equal("value", value);
     }
 
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyWithRedisInsight()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
+
+        using var builder = TestDistributedApplicationBuilder.CreateWithTestContainerRegistry(testOutputHelper);
+
+        var redis1 = builder.AddRedis("redis-1");
+        var redis2 = builder.AddRedis("redis-2").WithRedisInsight();
+
+        builder.Services.AddHttpClient();
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var factory = app.Services.GetRequiredService<IHttpClientFactory>();
+        var client = factory.CreateClient();
+
+        var redisInsightResource = builder.Resources.OfType<RedisInsightResource>().SingleOrDefault();
+
+        var insightEndpoint = redisInsightResource!.PrimaryEndpoint;
+
+        var getDatabasesApiUrl = $"{insightEndpoint.Scheme}://{insightEndpoint.Host}:{insightEndpoint.Port}/api/databases";
+
+        var pipeline = new ResiliencePipelineBuilder().AddRetry(new Polly.Retry.RetryStrategyOptions
+        {
+            Delay = TimeSpan.FromSeconds(2),
+            MaxRetryAttempts = 10,
+        }).Build();
+
+        await pipeline.ExecuteAsync(async (ct) =>
+        {
+            var response = await client.GetAsync(getDatabasesApiUrl, ct)
+                .ConfigureAwait(false);
+            response.EnsureSuccessStatusCode();
+
+            var databases = await response.Content.ReadFromJsonAsync<List<RedisInsightDatabaseModel>>(ct);
+
+            Assert.NotNull(databases);
+            Assert.Collection(databases,
+            db =>
+            {
+                Assert.Equal(redis1.Resource.Name, db.Name);
+                Assert.Equal(redis1.Resource.PrimaryEndpoint.ContainerHost, db.Host);
+                Assert.Equal(redis1.Resource.PrimaryEndpoint.Port, db.Port);
+                Assert.Equal("STANDALONE", db.ConnectionType);
+                Assert.Equal(0, db.Db);
+            },
+            db =>
+            {
+                Assert.Equal(redis2.Resource.Name, db.Name);
+                Assert.Equal(redis2.Resource.PrimaryEndpoint.ContainerHost, db.Host);
+                Assert.Equal(redis2.Resource.PrimaryEndpoint.Port, db.Port);
+                Assert.Equal("STANDALONE", db.ConnectionType);
+                Assert.Equal(0, db.Db);
+            });
+
+            foreach (var db in databases)
+            {
+                var testDatabaseConnectionApiUr = $"{insightEndpoint.Scheme}://{insightEndpoint.Host}:{insightEndpoint.Port}/api/databases/test/{db.Id}";
+                var cts = new CancellationTokenSource(TimeSpan.FromSeconds(2));
+                var testConnectionResponse = await client.GetAsync(getDatabasesApiUrl, cts.Token)
+                    .ConfigureAwait(false);
+                response.EnsureSuccessStatusCode();
```

Comment:
If this fails then it would be useful to throw an exception with the url that failed.